doc/source/rllib/rllib-offline.rst

```diff
@@ -5,437 +5,2081 @@
+        # need to make some further transformations specific for your data or
+        # loss. The default is `None`` which uses the base `OfflinePreLearner`
+        # defined in `ray.rllib.offline.offline_prelearner`. 
+        prelearner_class: Optional[Type],
+        # An optional `EpisodeReplayBuffer` class is
+        # used to buffer experiences when data is in `EpisodeType` or
+        # RLlib's previous `SampleBatch` type format. In this case, a single
+        # data row may contain multiple timesteps and the buffer serves two
+        # purposes: (a) to store intermediate data in memory, and (b) to ensure
+        # that exactly `train_batch_size_per_learner` experiences are sampled
+        # per batch. The default is RLlib's `EpisodeReplayBuffer`.
+        prelearner_buffer_class: Optional[Type],
+        # Optional keyword arguments for intializing the
+        # `EpisodeReplayBuffer`. In most cases this is simply the `capacity`
+        # for the default buffer used (`EpisodeReplayBuffer`), but it may
+        # differ if the `prelearner_buffer_class` uses a custom buffer.
+        prelearner_buffer_kwargs: Optional[Dict],
+        # Number of updates to run in each learner
+        # during a single training iteration. If None, each learner runs a
+        # complete epoch over its data block (the dataset is partitioned into
+        # at least as many blocks as there are learners). The default is `None`.
+        # This must be set to `1`, if a single (local) learner is used.
+        dataset_num_iters_per_learner: Optional[int],
+    )
 
 Output API
 ----------
 
 You can configure experience output for an agent using the following options:
 
-.. tip::
-    Plain python config dicts will soon be replaced by :py:class:`~ray.rllib.algorithms.algorithm_config.AlgorithmConfig`
-    objects, which have the advantage of being type safe, allowing users to set different config settings within
-    meaningful sub-categories (e.g. ``my_config.offline_data(input_=[xyz])``), and offer the ability to
-    construct an Algorithm instance from these config objects (via their ``.build()`` method).
-
 .. code-block:: python
 
-    # Specify where experiences should be saved:
-    #  - None: don't save any experiences
-    #  - "logdir" to save to the agent log dir
-    #  - a path/URI to save to a custom output directory (e.g., "s3://bucket/")
-    #  - a function that returns a rllib.offline.OutputWriter
-    "output": None,
-    # Arguments accessible from the IOContext for configuring custom output
-    "output_config": {},
-    # What sample batch columns to LZ4 compress in the output data.
-    "output_compress_columns": ["obs", "new_obs"],
-    # Max output file size (in bytes) before rolling over to a new file.
-    "output_max_file_size": 64 * 1024 * 1024,
-
-The interface for a custom output writer is as follows:
-
-.. autoclass:: ray.rllib.offline.OutputWriter
-    :members:
-    :noindex:
+    def offline_data(
```

Comment:
Ah, I see. You wanted to split this into in- and output. Ok.

Hmm, if it's that distinguishable, maybe we should separate this into 2 methods?

Ok, let's discuss offline, before we make a move here.